src/Compilers/CSharp/Portable/Compiler/Instrumentation.cs

```diff
@@ -30,34 +30,40 @@ internal static BoundBlock InjectInstrumentation(MethodSymbol method, BoundBlock
                     TypeSymbol boolType = factory.SpecialType(SpecialType.System_Boolean);
                     TypeSymbol payloadElementType = boolType;
                     ArrayTypeSymbol payloadType = ArrayTypeSymbol.CreateCSharpArray(compilation.Assembly, payloadElementType);
-                    FieldSymbol payloadField = GetPayloadField(method, methodOrdinal, payloadType, factory);
                     bool methodHasExplicitBlock = MethodHasExplicitBlock(method);
-
+                    LocalSymbol methodPayload = factory.SynthesizedLocal(payloadType, kind: SynthesizedLocalKind.InstrumentationPayload, syntax: methodBody.Syntax);
+                    ArrayTypeSymbol modulePayloadType = ArrayTypeSymbol.CreateCSharpArray(compilation.Assembly, payloadType);
+                    // PROTOTYPE (https://github.com/dotnet/roslyn/issues/10411): In the future there will be multiple analysis kinds.
+                    int analysisKind = 0;
                     // Synthesize the instrumentation and collect the spans of interest.
 
+                    PrivateImplementationDetails privateImplementationDetails = compilationState.ModuleBuilderOpt.GetPrivateImplClass(methodBody.Syntax, diagnostics);
```

Comment:
The need for it apparently came and went during development, and I neglected to remove this local. Will remove.